File: backend/katzenpost/src/main/java/com/fsck/k9/backend/katzenpost/DummyFolder.java
package com.fsck.k9.backend.katzenpost;


import java.io.IOException;
import java.util.Date;
import java.util.List;
import java.util.Map;
import java.util.Set;

import com.fsck.k9.mail.FetchProfile;
import com.fsck.k9.mail.Flag;
import com.fsck.k9.mail.Folder;
import com.fsck.k9.mail.Message;
import com.fsck.k9.mail.MessageRetrievalListener;
import com.fsck.k9.mail.MessagingException;


public class DummyFolder extends Folder<Message> {
    @Override
    public void open(int mode) throws MessagingException {

    }

    @Override
    public void close() {

    }

    @Override
    public boolean isOpen() {
        return false;
    }

    @Override
    public int getMode() {
        return 0;
    }

    @Override
    public boolean create(FolderType type) throws MessagingException {
        return false;
    }

    @Override
    public boolean exists() throws MessagingException {
        return false;
    }

    @Override
    public int getMessageCount() throws MessagingException {
        return 0;
    }

    @Override
    public int getUnreadMessageCount() throws MessagingException {
        return 0;
    }

    @Override
    public int getFlaggedMessageCount() throws MessagingException {
        return 0;
    }

    @Override
    public Message getMessage(String uid) throws MessagingException {
        return null;
    }

    @Override
    public boolean areMoreMessagesAvailable(int indexOfOldestMessage, Date earliestDate)
            throws IOException, MessagingException {
        return false;
    }

    @Override
    public Map<String, String> appendMessages(List<? extends Message> messages) throws MessagingException {
        return null;
    }

    @Override
    public void setFlags(List<? extends Message> messages, Set<Flag> flags, boolean value) throws MessagingException {

    }

    @Override
    public void setFlags(Set<Flag> flags, boolean value) throws MessagingException {

    }

    @Override
    public String getUidFromMessageId(String messageId) throws MessagingException {
        return null;
    }

    @Override
    public void delete(boolean recurse) throws MessagingException {

    }

    @Override
    public String getServerId() {
        return null;
    }

    @Override
    public String getName() {
        return null;
    }

    @Override
    public void fetch(List<Message> messages, FetchProfile fp, MessageRetrievalListener<Message> listener)
            throws MessagingException {

    }

    @Override
    public List<Message> getMessages(int start, int end, Date earliestDate, MessageRetrievalListener<Message> listener)
            throws MessagingException {
        return null;
    }
}

File: backend/katzenpost/src/main/java/com/fsck/k9/backend/katzenpost/DummyFolder.kt
package com.fsck.k9.backend.katzenpost


import java.io.IOException
import java.util.Date

import com.fsck.k9.mail.FetchProfile
import com.fsck.k9.mail.Flag
import com.fsck.k9.mail.Folder
import com.fsck.k9.mail.Message
import com.fsck.k9.mail.MessageRetrievalListener
import com.fsck.k9.mail.MessagingException


class DummyFolder(val folderName: String) : Folder<Message>() {
    var openMode: Int? = null

    override fun getName() = folderName

    @Throws(MessagingException::class)
    override fun open(mode: Int) {
        openMode = mode
    }

    override fun close() {
        openMode = null
    }

    override fun isOpen() = openMode != null

    override fun getMode(): Int {
        return openMode ?: 0
    }

    @Throws(MessagingException::class)
    override fun create(type: Folder.FolderType): Boolean {
        return true
    }

    @Throws(MessagingException::class)
    override fun exists(): Boolean {
        return true
    }

    @Throws(MessagingException::class)
    override fun getMessageCount(): Int {
        return 0
    }

    @Throws(MessagingException::class)
    override fun getUnreadMessageCount(): Int {
        return 0
    }

    @Throws(MessagingException::class)
    override fun getFlaggedMessageCount(): Int {
        return 0
    }

    @Throws(MessagingException::class)
    override fun getMessage(uid: String): Message? {
        return null
    }

    @Throws(IOException::class, MessagingException::class)
    override fun areMoreMessagesAvailable(indexOfOldestMessage: Int, earliestDate: Date): Boolean {
        return false
    }

    @Throws(MessagingException::class)
    override fun appendMessages(messages: List<Message>): Map<String, String>? {
        return null
    }

    @Throws(MessagingException::class)
    override fun setFlags(messages: List<Message>, flags: Set<Flag>, value: Boolean) {

    }

    @Throws(MessagingException::class)
    override fun setFlags(flags: Set<Flag>, value: Boolean) {

    }

    @Throws(MessagingException::class)
    override fun getUidFromMessageId(messageId: String): String? {
        return null
    }

    @Throws(MessagingException::class)
    override fun delete(recurse: Boolean) {

    }

    override fun getServerId(): String? {
        return null
    }

    @Throws(MessagingException::class)
    override fun fetch(messages: List<Message>, fp: FetchProfile, listener: MessageRetrievalListener<Message>) {

    }

    @Throws(MessagingException::class)
    override fun getMessages(start: Int, end: Int, earliestDate: Date, listener: MessageRetrievalListener<Message>): List<Message>? {
        return emptyList()
    }
}
